[custom-build] sql/rowexec: subject column backfills to admission control

erikgrinaker opened the pull request:
Temporary 21.2 backport of #79115 and friends, for a custom build.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79117)
<!-- Reviewable:end -->

ajwerner committed:
pkg/util/admission: track history for bytes added per work

Before this patch, when no bytes were admitted in an interval, we'd assume that
the correct estimate per byte was 1. This meant that we could see a huge jump
in tokens which effectively disabled the throttling of admission control.

Release note (bug fix): Fixed a bug in IO admission control which could result
in admission control failing to rate-limit when traffic was stalled such that
no work was admitted, despite the store's being in an unhealthy state.

dt committed:
sql/importer: re-validate unique constraints after IMPORT

Release note (bug fix): Previously IMPORT INTO could create duplicate entries UNIQUE constraints in REGIONAL BY ROW tables and tables utilizing UNIQUE WITHOUT INDEX constraints. A new post-IMPORT validation step for those tables now fails and rolls back the IMPORT in such cases.

ajwerner committed:
vendor: pick up hot fix for compaction

https://github.com/cockroachdb/pebble/pull/1625

Release note: None
